File: Programmers/level1/모의고사.py
def solution(answers):
    score=[0,0,0]
    person=[[1, 2, 3, 4, 5],
            [ 2, 1, 2, 3, 2, 4, 2, 5],
            [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    
    for p in range(3):
        result=[1 for idx, val in enumerate(answers) if val==person[p][idx%len(person[p])]]
        score[p]=sum(result)
    
    m=max(score)
    answer=[idx+1 for idx, val in enumerate(score) if m==val ]
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    static int[][] pattern={
        {1,2,3,4,5},
        {2,1,2,3,2,4,2,5},
        {3,3,1,1,2,2,4,4,5,5}
    };
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] cnt=new int[3];
        int max=0;
        int ansSize=0;
        for(int i=0;i<answers.length;i++){
            for(int j=0;j<3;j++){
                if(answers[i]==pattern[j][i%pattern[j].length]){
                    cnt[j]++;
                    if(max<cnt[j]){
                        ansSize=1;
                        max=cnt[j];
                    }else if(max==cnt[j]) ansSize++;
                }
                
            }
        }
        
        answer=new int[ansSize];
        int idx=0;
        for(int i=0;i<3;i++){
            if(cnt[i]==max) answer[idx++]=i+1;
        }
        return answer;
    }
}